sequence_processing_pipeline/Pipeline.py

```diff
@@ -201,6 +215,68 @@ def validate(self, sample_sheet_path):
 
         return msgs, val_sheet
 
+    def generate_sample_information_files(self, sample_sheet_path):
+        '''
+        Using a path to a validated sample-sheet, generate sample-information
+        files in self.output_path.
+        :param sample_sheet_path:
+        :return: A list of paths to sample-information-files.
+        '''
+        sheet = KLSampleSheet(sample_sheet_path)
+        msgs, val_sheet = quiet_validate_and_scrub_sample_sheet(sheet)
+
+        if val_sheet is None:
+            raise PipelineError("'%s' is not a valid sample-sheet." %
+                                sample_sheet_path)
+
+        samples = []
+        for sample in val_sheet.samples:
+            if sample['Sample_ID'].startswith('BLANK'):
+                samples.append((sample['Sample_ID'], sample['Sample_Project']))
+
+        projects = list(set([y for x, y in samples]))
+
+        paths = []
+        for project in projects:
+            samples_in_proj = [x for x, y in samples if y == project]
+            some_path = join(self.output_path, f'{project}_blanks.tsv')
+            paths.append(some_path)
+            with open(some_path, 'w') as f:
+                # write out header to disk
+                f.write('\t'.join(Pipeline.sif_header) + '\n')
+
+                # for now, populate values that can't be derived from the
+                # sample-sheet w/'EMPTY'.
+                for sample in samples_in_proj:
+                    row = {}
+                    for column, default_value in zip(Pipeline.sif_header,
+                                                     Pipeline.sif_defaults):
+                        # ensure all defaults are converted to strings.
+                        row[column] = str(default_value)
+
+                    # generate values for the four columns that must be
+                    # determined from sample-sheet information.
+
+                    # convert 'BLANK14_10F' to 'BLANK14.10F', etc.
+                    row['sample_name'] = sample.replace('_', '.')
+                    row['host_subject_id'] = sample.replace('_', '.')
+                    row['description'] = sample.replace('_', '.')
+
+                    # generate collection_timestamp from self.run_id
+                    # assume all run_ids begin with coded datestamp:
+                    # 210518_...
+                    # allow exception if substrings cannot convert to int
+                    # or if array indexes are out of bounds.
+                    year = int(self.run_id[0:2]) + 2000
```

Comment:
Are you sure that the first two digits are the year and not the day?

Reply:
I believe so, after looking at several recent run_ids like, '211201_MN01225_0016_A000H32T2K'. In this case it seems like the only correct interpretation would be 21 for year and 12 for month and 01 for day.